packages/Chem/src/panels/structural-alerts.ts

```diff
@@ -1,69 +1,37 @@
-// This file may not be used in
-import * as ui from 'datagrok-api/ui';
-import * as grok from 'datagrok-api/grok';
 import * as DG from 'datagrok-api/dg';
-import {getRdKitModule, getRdKitWebRoot} from '../utils/chem-common-rdkit';
-import {RDModule, RDMol} from '@datagrok-libraries/chem-meta/src/rdkit-api';
 
-export async function checkForStructuralAlerts(col: DG.Column<string>): Promise<void> {
-  if (col.length > 500) {
-    grok.shell.info('Could not check structural alerts: too much data (max. 500 rows)');
-    return;
+import {RDModule, RDMol} from '@datagrok-libraries/chem-meta/src/rdkit-api';
+import BitArray from '@datagrok-libraries/utils/src/bit-array';
+import { getRdKitService } from '../utils/chem-common-rdkit';
+import { RdKitService } from '../rdkit-service/rdkit-service';
+
+export type RuleId = 'PAINS' | 'BMS' | 'SureChEMBL' | 'MLSMR' | 'Dandee' | 'Inpharmatica' | 'LINT' | 'Glaxo';
+export type RuleSet = {[rule in RuleId]: boolean};
+
+export async function runStructuralAlertsDetection(moleculeCol: DG.Column<string>, ruleSet: RuleSet, alertsDf: DG.DataFrame,
+  rdkitService: RdKitService): Promise<DG.DataFrame> {
+  const ruleSetCol: DG.Column<string> = alertsDf.getCol('rule_set_name');
+  const ruleSetColCategories = ruleSetCol.categories;
+  const ruleSetColData = ruleSetCol.getRawData();
+  const smartsCol: DG.Column<string> = alertsDf.getCol('smarts');
```

Comment:
Does ruleSetColumn contain strings?
https://datagrok.ai/js-api/classes/dg.BigIntColumn#getrawdata specifies the usage for Int32Array | Uint32Array | Float32Array | Float64Array. How it is supposed to be used here?